Christmas Eve should not be included in calculations

For e.g. PCCs, SI 2012/1917, Schedule 3, Part 1, section 3(b) excludes Christmas Eve as a day when calculating dates, but the code does not handle this. e.g. an election date of 2019-01-10 should give a SOPN date of 2018-12-11, but currently gives a date of 2018-12-12.

```
============================================================ FAILURES =============================================================
_________________________________________ test_january_date_police_and_crime_commissioner _________________________________________
    def test_january_date_police_and_crime_commissioner():
        publish_date = sopn_publish_date.police_and_crime_commissioner(date(2019, 1, 10))

>       assert publish_date == date(2018, 12, 11)
E       assert datetime.date(2018, 12, 12) == datetime.date(2018, 12, 11)
E         -datetime.date(2018, 12, 12)
E         +datetime.date(2018, 12, 11)

tests/test_sopn_publish_date.py:136: AssertionError
```

It looks like the same applies to all the other ones I've checked as well.

--- a/sopn_publish_date/calendars.py
+++ b/sopn_publish_date/calendars.py
@@ -62,9 +62,15 @@ class UKBankHolidayCalendar(AbstractHolidayCalendar):
     def __init__(self, dates):
         AbstractHolidayCalendar.__init__(self)
 
-        self.rules = [
+        christmas_eve = Holiday("Christmas Eve", month=12, day=24)
+
+        days_not_counted = [
             UKBankHolidayCalendar.create_holiday_from_entry(entry) for entry in dates
         ]
+
+        days_not_counted.append(christmas_eve)
+
+        self.rules = days_not_counted
 
 
 class GibraltarBankHolidays(BankHolidayCalendar):

--- a/tests/test_sopn_publish_date.py
+++ b/tests/test_sopn_publish_date.py
@@ -216,3 +216,17 @@ def test_publish_date_uk_parliament_2019():
     publish_date = sopn_publish_date.uk_parliament(date(2019, 12, 12))
 
     assert publish_date == date(2019, 11, 14)
+
+
+def test_christmas_eve_not_counted():
+
+    election_and_expected_sopn_date = {
+        sopn_publish_date.police_and_crime_commissioner: date(2018, 12, 11),
+        sopn_publish_date.uk_parliament: date(2018, 12, 7),
+        sopn_publish_date.scottish_parliament: date(2018, 12, 3),
+        sopn_publish_date.northern_ireland_assembly: date(2018, 12, 13),
+        sopn_publish_date.national_assembly_for_wales: date(2018, 12, 10),
+    }
+
+    for sopn_for_election_on, expected_date in election_and_expected_sopn_date.items():
+        assert sopn_for_election_on(date(2019, 1, 10)) == expected_date

